rs/nns/cmc/src/main.rs

```diff
@@ -442,8 +443,15 @@ fn set_authorized_subnetwork_list(arg: SetAuthorizedSubnetworkListArgs) {
     with_state_mut(|state| {
         let governance_canister_id = state.governance_canister_id;
 
-        if CanisterId::unchecked_from_principal(caller()) != governance_canister_id {
-            panic!("Only the governance canister can set authorized subnetwork lists.");
+        let caller_id = CanisterId::unchecked_from_principal(caller());
+        let subnet_rental_canister_id = CanisterId::unchecked_from_principal(
+            PrincipalId::from_str("qvhpv-4qaaa-aaaaa-aaagq-cai").unwrap(),
+        );
+
+        if !(caller_id == governance_canister_id || caller_id == subnet_rental_canister_id) {
```

Comment:
This should be more like

```suggestion
        let is_authorized = SET_AUTHORIZED_SUBNETWORK_AUTHORIZED_CALLERS.contains(caller_id);
        if !is_authorized {
```

Reply:
I think if we don't foresee more canisters calling this, I'd refrain from creating a list of allowed canisters and just have the two canisters there. I've updated it to use an already present however.